docs/design-guide/neutron-networking.md

```diff
@@ -0,0 +1,339 @@
+- [ovn-router][ovn-admin] as the L3 router plugin
+- [trunk plugin][neutron-trunk] service plugin
+- [network segment range][neutron-network-segment-range] service plugin
+
+The physical network design for each site is a leaf/spine configuration
+so to best support this we use the VXLAN type driver with VLANs on all the
+leaves bound to the VXLAN VNI. In this configuration there is only one subnet
+across the network, in Neutron this is called [L2 Adjacency][neutron-l2-adjacency].
+This model has been used by networking vendors such as Arista, Juniper, and Cisco
+in their own ML2 mechanism driver. However full support for this has been lacking
+upstream so we have developed our own mechanism driver as we explore the best
+approach. Traditionally Ironic (baremetal) based Neutron deployment have utilized
+the [networking-generic-switch][networking-generic-switch] mechanism however as it
+stands today it only supports VLAN and it's templating capabilities are not sufficient
+for our needs. We hope to eventually create a generic mechanism which can be contributed
+back. Another limitation that we are aware of is in the VXLAN type driver itself,
+the Neutron team designed this type as an overlay style VXLAN system while we are
+focused on an underlay or EVPN-VXLAN style system. We are actively working with
+upstream on how to best include this use case.
+
+## Mapping Leaf/Spine to Neutron Networks
+
+The Leaf/Spine network is implemented as a VNI that exists on the fabric with
+endpoints on each of the leaves that need connectivity being mapped to a VLAN
+to provide connectivity to physical assets connected to that leaf.
+
+### VNIs
+
+To provide available VNIs a [network segment range][neutron-network-segment-range] is
+created of type VXLAN with the name of the fabric.
+
+<!-- markdownlint-capture -->
+<!-- markdownlint-disable MD046 -->
+!!! Note
+
+    This is one of the places that the underlay vs overlay VXLAN disconnect
+    rears its head. Neutron assumes that all VXLAN participants will be able
+    to reach each other via the tunnel configuration. However it is possible
+    to have multiple physically separate fabrics but Neutron does not allow
+    the VXLAN type to have a `physical_network` value like VLAN networks.
+<!-- markdownlint-restore -->
+
+Provider networks and self-service networks are allocated VNIs from this range.
+
+### VLANs
+
+For each leaf pair on the fabric a VLAN [network segment range][neutron-network-segment-range]
+is created with the `physical_network` value set to their pair name. For example,
+the name of the rack they serve could be used.
+
+### Connecting a Server to a Network
+
+When a server needs to have a connection to a network, it's `local_link_connection`
+information is looked up. The `physical_network` is then used to attempt to map
+the port to how it connects on the network. This is documented in Ironic's
+Networking Guide as [VIF Attachment][ironic-vif-attachment].
```

Comment:
When a server needs to establish a connection to a network, Ironic checks the baremetal port's `local_link_connection` and `physical_network` attributes to determine how it should be mapped to the desired network. This information is documented in Ironic's Networking Guide as [VIF Attachment][ironic-vif-attachment].